datafusion/physical-expr/src/aggregate/count_distinct.rs

```diff
@@ -216,23 +253,27 @@ impl Accumulator for DistinctCountAccumulator {
     }
 
     fn size(&self) -> usize {
-        std::mem::size_of_val(self)
-            + (std::mem::size_of::<DistinctScalarValues>() * self.values.capacity())
-            + self
-                .values
-                .iter()
-                .map(|vals| {
-                    ScalarValue::size_of_vec(&vals.0) - std::mem::size_of_val(&vals.0)
-                })
-                .sum::<usize>()
-            + (std::mem::size_of::<DataType>() * self.state_data_types.capacity())
-            + self
-                .state_data_types
-                .iter()
-                .map(|dt| dt.size() - std::mem::size_of_val(dt))
-                .sum::<usize>()
-            + self.count_data_type.size()
-            - std::mem::size_of_val(&self.count_data_type)
+        match &self.count_data_type {
+            DataType::Boolean
+            | DataType::Date32
+            | DataType::Date64
+            | DataType::Float16
+            | DataType::Float32
+            | DataType::Float64
+            | DataType::Int16
+            | DataType::Int32
+            | DataType::Int64
+            | DataType::Int8
+            | DataType::Time32(_)
+            | DataType::Time64(_)
+            | DataType::Null
+            | DataType::Timestamp(_, _)
```

Comment:
I think this is missing some types like `Interval`  and `Duration`

Perhaps you could check instead if `DataType::primitive_width` returned `Some(.)` 🤔 

https://docs.rs/arrow/34.0.0/arrow/datatypes/enum.DataType.html#method.primitive_width

Reply:
Done, seems in new arrow-rs @tustvold made some work for us, and introduced `Datatype::is_primitive`